src/d3d11/d3d11_main.cpp

```diff
@@ -14,7 +14,39 @@ namespace dxvk {
   
 extern "C" {
   using namespace dxvk;
-  
+
+  //
+  // Workaround MinGW symbol name issues that prevent using DXVK in Wine with a 32-bit
+  // mod that works by proxying dxgi.dll.
+  //
+  // If a linker-dependency on DXVK's dxgi.dll were used, then it would require DXGI.dll
+  // to export "CreateDXGIFactory1@8" and Windows mod DLLs do not do this.
+  //
+  //  Mods can make themselves compatible with old versions of DXVK by adding this code:
+  // 
+  //   #ifdef _M_IX86
+  //   #pragma comment (linker, "/export:CreateDXGIFactory1@8=_CreateDXGIFactory1@8")
+  //   #endif
+  //
+  // To support unaltered 32-bit game mods, DXVK must load the DLL manually...
+  //
+  HRESULT __stdcall CreateDXGIFactory1_Import(REFIID riid, void **ppFactory) {
+#ifdef _M_IX86
+    using CreateDXGIFactory1_pfn = HRESULT (WINAPI*)(REFIID,void**);
+          CreateDXGIFactory1_pfn
+       DllCreateDxgiFactory1 =
+         (CreateDXGIFactory1_pfn)GetProcAddress(LoadLibraryA("dxgi.dll"),
```

Comment:
Could we just do GetModuleHandle to not leak a reference?

Probably not a big deal since ideally this should be cleaned up when unloading d3d11.dll anyway, but you never know.

Reply:
GetModuleHandle would not handle the case where `d3d11.dll` is manually loaded by an application and the application has not loaded `dxgi.dll` yet.

`GetModuleHandle (...)` would only work if D3D11.dll has a linker dependency on dxgi.dll, and that's what's trying to be avoided here. `LoadLibrary` is the only way to replicate the normal DLL loader behavior where DXVK's d3d11.dll loads dxgi.dll automatically.

I'm not sure how this would constitute a reference leak? If we went the linker dependency route, d3d11.dll would be holding a reference to dxgi.dll the entire time it's loaded, so kind of status quo. I guess if d3d11.dll is expected to be unloaded while the program continues running, then it could be a problem, but does that ever actually happen in the real-world?

Reply:
Hm, yeah, if we don't get the DLL loaded at all then that won't work.